packages/web/components/complex/all-pools-table-set.tsx

```diff
@@ -0,0 +1,154 @@
+import { observer } from "mobx-react-lite";
+import { FunctionComponent, useState } from "react";
+import { useAllPoolsTable } from "../../hooks/use-all-pools-table";
+import { useStore } from "../../stores";
+import { CheckBox, MenuToggle, PageList, SortMenu } from "../control";
+import { SearchBox } from "../input";
+import { Table } from "../table";
+import { MetricLoaderCell, PoolCompositionCell } from "../table/cells";
+
+const poolsMenuOptions = [
+  { id: "incentivized-pools", display: "Incentivized Pools" },
+  { id: "all-pools", display: "All Pools" },
+];
+
+const TVL_FILTER_THRESHOLD = 1000;
+
+export const AllPoolsTableSet: FunctionComponent = observer(() => {
+  const {
+    chainStore,
+    queriesExternalStore,
+    priceStore,
+    queriesOsmosisStore,
+    accountStore,
+  } = useStore();
+  const [activeOptionId, setActiveOptionId] = useState(poolsMenuOptions[0].id);
+  const [isPoolTvlFiltered, setIsPoolTvlFiltered] = useState(false);
+
+  const chainInfo = chainStore.osmosis;
+  const queryOsmosis = queriesOsmosisStore.get(chainInfo.chainId);
+  const queryExternal = queriesExternalStore.get();
+  const account = accountStore.getAccount(chainInfo.chainId);
+
+  const allPools = queryOsmosis.queryGammPools.getAllPools();
+  const incentivizedPoolIds =
+    queryOsmosis.queryIncentivizedPools.incentivizedPools;
+
+  const allPoolsWithMetrics = allPools.map((pool) => ({
+    ...queryExternal.queryGammPoolFeeMetrics.makePoolWithFeeMetrics(
+      pool,
+      priceStore
+    ),
+    myLiquidity: pool
+      .computeTotalValueLocked(priceStore)
+      .mul(
+        queryOsmosis.queryGammPoolShare.getAllGammShareRatio(
+          account.bech32Address,
+          pool.id
+        )
+      ),
+    apr: queryOsmosis.queryIncentivizedPools.computeMostAPY(
+      pool.id,
+      priceStore
+    ),
+  }));
+  const incentivizedPoolsWithMetrics = allPoolsWithMetrics.reduce(
+    (
+      incentivizedPools: ObservablePoolWithFeeMetrics[],
+      poolWithMetrics: ObservablePoolWithFeeMetrics
+    ) => {
+      if (
+        incentivizedPoolIds.some(
+          (incentivizedPoolId) => poolWithMetrics.pool.id === incentivizedPoolId
+        )
+      ) {
+        incentivizedPools.push(poolWithMetrics);
+      }
+      return incentivizedPools;
+    },
+    []
+  );
+
+  const isIncentivizedPools = activeOptionId === poolsMenuOptions[0].id;
+  const activeOptionPools = isIncentivizedPools
+    ? incentivizedPoolsWithMetrics
+    : allPoolsWithMetrics;
+  const tvlFilteredPools = isPoolTvlFiltered
+    ? activeOptionPools
+    : activeOptionPools.filter((poolWithMetrics) =>
+        poolWithMetrics.liquidity.toDec().gte(new Dec(TVL_FILTER_THRESHOLD))
+      );
+
+  const {
+    query,
+    setQuery,
+    sortKeyPath,
+    setSortKeyPath,
+    toggleSortDirection,
+    page,
+    setPage,
+    minPage,
+    numPages,
+    tableCols,
+    tableRows,
+    tableData,
+  } = useAllPoolsTable(tvlFilteredPools, isIncentivizedPools);
```

Comment:
This may not be what you want to hear but I actually think both of these hooks should have their contents put in these `/complex/` component files. I think with the high number of output params, it's a sign that it's not worth containing within it's own componenet. If we have the code in these files with comments to describe the high level of what is happening in sections of the code then I think we've accomplished the same thing.